pkg/resource/v1/certconfig/update.go

```diff
@@ -11,5 +14,54 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange inte
 }
 
 func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*framework.Patch, error) {
-	return nil, nil
+	create, err := r.newCreateChange(ctx, obj, currentState, desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	update, err := r.newUpdateChange(ctx, obj, currentState, desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	patch := framework.NewPatch()
+	patch.SetCreateChange(create)
+	patch.SetUpdateChange(update)
+
+	return patch, nil
+}
+
+func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desiredState interface{}) ([]*v1alpha1.CertConfig, error) {
+	currentCertConfigs, err := toCertConfigs(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	desiredCertConfigs, err := toCertConfigs(desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "debug", "finding out which certconfigs have to be updated")
+
+	var certConfigsToUpdate []*v1alpha1.CertConfig
+	for _, currentCertConfig := range currentCertConfigs {
+		desiredCertConfig, err := getCertConfigByName(desiredCertConfigs, currentCertConfig.Name)
+		if IsNotFound(err) {
+			// NOTE that this case indicates we should remove the current
+			// certconfig eventually.
+			r.logger.LogCtx(ctx, "warning", fmt.Sprintf("not updating certconfig '%s': no desired certconfig found", currentCertConfig.GetName()))
```

Comment:
Question about general understanding: This means that there is a certconfig which is not part of our desired state, right? I believe we could alert on this, as we have a mismatch and possibly orphaned resources.

Reply:
Sorry I was imprecise here, I thought that the operator should return a metric on this orphaned metric. AWS-operator does this for orphaned resources. This should not be part of this PR or anything, so you're fine here. I was just wondering if I understood the flow correctly. All good :+1: